%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Design based on a template by Roberto and following the format of
% the xmipp tutorials. In turn, they seem to be based on a template
% from http://www.latextemplates.com
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%	PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------

\documentclass[12pt]{article} % Default font size is 12pt, it can be changed here
\usepackage[english]{babel}
\usepackage[utf8]{inputenc}
\usepackage{listings} % To include source code
\usepackage{caption}
\usepackage[htt]{hyphenat}
\usepackage{geometry} % Required to change the page size to A4
%\geometry{a4paper} % Set the page size to be A4 as opposed to the default US Letter
\usepackage{framed}
\usepackage{url}
\usepackage{graphicx} % Required for including pictures
\usepackage{natbib}
\usepackage{float} % Allows putting an [H] in \begin{figure} to specify the exact location of the figure
%\usepackage{hyperref}
\usepackage{menukeys}
\usepackage{array}
\usepackage{fancyhdr}
\usepackage{marvosym}%smileys\Smiley{} \Frowny{}
\usepackage{etoolbox}
\usepackage{listings}
\usepackage{makecell}
\usepackage{marginnote}
\usepackage{soul}
\usepackage[toc,page]{appendix}
\usepackage{caption}
%\usepackage{menukeys}
\usepackage{fancybox,framed}
\usepackage{xspace}
\usepackage{rotating}
\usepackage{gensymb}
\usepackage{vhistory}

%commands
\newcommand{\ffigure}[1]{{Fig. {\ref{#1}}}\xspace}
\newcommand{\ttable}[1]{{Table {\ref{#1}}}\xspace}
\newcommand{\scommand}[1]{{{\keys{#1}}}\xspace}
%definitions
\def\ccmask{CC\textsubscript{MASK}\xspace}
\def\ccp4{\textit{CCP4}\xspace}
\def\chimera{\textit{ChimeraX}\xspace}
\def\coot{\textit{Coot}\xspace}
\def\emringer{\textit{EMRinger}\xspace}
\def\modeller{\textit{Modeller}\xspace}
\def\molprobity{\textit{MolProbity}\xspace}
\def\validationCryoEM{\textit{Validation CryoEM}\xspace}
\def\phenix{\textit{PHENIX}\xspace}
\def\powerfit{\textit{PowerFit}\xspace}
\def\refmac{\textit{Refmac}\xspace}
\def\scipion{\textit{Scipion}\xspace}

\sethlcolor{yellow}

%\renewcommand{\hl}[1]{#1}
%pdflatex -jobname=students '\def\student{}\input{main}'
%pdflatex -jobname=teachers '\def\teachers{}\input{main}'
%  \ifdef{\teachers}
%  {Content for teachers}
%  {Content for students} 
\newcommand{\ttt}[1]{\texttt{#1}}
\newcommand{\iii}[1]{\textit{#1}}
\newcommand{\ra}{$\rightarrow$}
\pagestyle{fancy}
\fancyhf{}
\fancyhead[RO]{{Model Building}}
\fancyhead[LO]{Scipion}
%\fancyhead[RO]{{\leftmark}}
\fancyfoot[RO]{\thepage}

\linespread{1.2} % Line spacing

%\setlength\parindent{0pt} % Uncomment to remove all indentation from paragraphs

\newenvironment{command}{\tt\begin{quote}}{\end{quote}}
\newcommand{\comm}[1]{\texttt{#1}}

\newcommand{\imgfig}[3]{\begin{figure}[H]\centering \
\includegraphics[scale=#2]{images/#1} \caption{#3} \end{figure}}

\newcommand{\proto}[1]{\textit{\textbf{#1}}}
\newcommand{\popt}[1]{\textit{#1}}
\newcommand{\pval}[1]{\texttt{#1}}

\newcommand\tstrut{\rule{0pt}{2.4ex}}
\newcommand\bstrut{\rule[-1.0ex]{0pt}{0pt}}

\def \humanAdenoMap {7034}%5172

\begin{document}

%----------------------------------------------------------------------------------------
%	TITLE PAGE
%----------------------------------------------------------------------------------------

\begin{titlepage}

% New command for horizontal lines. Change thickness here.
\newcommand{\HRule}{\rule{\linewidth}{0.5mm}}

\center % Center everything on the page

\includegraphics{images/scipion_logo}

{\large Scipion Tutorial Series}\\[1.0cm]

\textsc{\LARGE National Center for Biotechnology}\\[0.5cm]
\textsc{\Large Biocomputing Unit}\\[0.15cm]

\HRule\\[0.3cm]
{ \huge \bfseries Model Building Basic}\\ % Title of your document
\HRule \\[0.35cm]
{\large \today}\\ % Date, change the \today to a set date if you want to be precise
\begin{center}
\includegraphics[width=0.70\textwidth]{{images/aadensity}}\\
Density for amino acid side chains from an experimental electron density map at 1.5 \AA~resolution \footnotesize{(http://people.mbi.ucla.edu/sawaya/m230d/Modelbuilding/modelbuilding.html)}
\end{center}

%\vfill % Fill the rest of the page with whitespace
%\begin{minipage}{0.4\textwidth}
\begin{flushright}
 \large
%\emph{Author:}\\
  \textsc{Roberto Marabini \& Marta Mart√≠nez} % Your name
\end{flushright}
%\end{minipage}

\end{titlepage}

\begin{versionhistory}
  \vhEntry{1.0}{11.15.2018}{MM|RM}{created for first model building workshop}
  \vhEntry{1.1}{01.30.2019}{MM}{added appendices and minor fixes}
  \vhEntry{1.2}{04.24.2019}{MM}{added atomstructutils, contacts and submission protocols}
  \vhEntry{1.3}{09.10.2019}{MM}{added map preprocessing protocols (create mask, and compute local Resolution and Sharpening) and \phenix validation cryoEM}
  \vhEntry{1.4}{10.10.2020}{MM}{migration to python3 and adaptation to \scipion version 3.0, replacement of Chimera by ChimeraX, added \phenix dock-in-map protocol, removed singularities for \phenix version 1.13, removed \powerfit protocol}
\end{versionhistory}\newpage

%----------------------------------------------------------------------------------------
%	OBJETIVOS
%----------------------------------------------------------------------------------------


\subsection*{Intended audience}
The recent rapid development of single-particle electron cryo-microscopy (cryo-EM) allows structures to be solved by this method at almost atomic resolutions.  Providing a basic introduction to model building, this tutorial shows the initial workflow aimed at obtaining high-quality atomic models from cryo-EM data by using \scipion software framework. %tomography in  electron microscopy with special emphasis in basic image processing. The tutorial requires matlab but does not assume any programing skills. 


\subsection*{We'd like to hear from you}

We have tested and verified the different steps described in this demo
to the best of our knowledge, but since our programs are in continuous
development you may find inaccuracies and errors in this text. Please
let us know about any errors, as well as your suggestions for
future editions, by writing to
scipion@cnb.csic.es.


\subsection*{Requirements}

This tutorial requires, in addition to \scipion, USCF~\chimera (\url{https://www.rbvi.ucsf.edu/chimerax/download.html}), the \ccp4 suite (\url{http://www.ccp4.ac.uk/download/#os=linux}) including \refmac and \coot, and the \textit{PHENIX suite} (\url{https://www.phenix-online.org/download/}). Basic knowledge of \chimera and \scipion is assumed. Warning: old versions of \refmac are not suitable for EM data.

\newpage


%----------------------------------------------------------------------------------------
%	TABLE OF CONTENTS
%----------------------------------------------------------------------------------------

\tableofcontents % Include a table of contents

\newpage % Begins on a new page instead of on the same page as the table of contents


\input{010-introduction.tex}
\input{020-problem2solve.tex}
\input{030-inputDataDescription.tex}
\input{040-importInputData.tex}
\input{050-volumeScenario.tex}
\input{060-sequenceScenario.tex}
\input{070-movingFromSequence.tex}%%%
\input{080-rigidFit.tex}
\input{090-flexibleFit.tex}
\input{100-validation.tex}
\input{110-unitcell.tex}
\input{120-macromolecule.tex}
\input{121-submissionResults.tex}

\input{130-install.tex}
\input{140-todo.tex}

\bibliographystyle{elsart-harv}
\bibliography{../tutorial_common/em}


\begin{appendices}
\input{a010-answers.tex}
\input{a011-atomStructUtilsOperatorProtocol.tex}
\input{a012-chimeraContacts.tex}
\input{a020-chimeraOperate.tex}
\input{a030-chimeraRestoreSession.tex}
\input{a040-chimeraRigidFit.tex}
\input{a050-ccp4CootRefinement.tex}
\input{a060-ccp4Refmac.tex}
\input{a061-create3DMask.tex}
\input{a070-extractUnitCell.tex}
\input{a080-importAtomicStructure.tex}
\input{a090-importSequence.tex}
\input{a091-importMask.tex}
\input{a100-importvolume.tex}
\input{a101-localDeblurSharpening.tex}
\input{a102-localMonoRes.tex}
\input{a103-deepEMhancerSharpening.tex}
\input{a110-modelFromTemplate.tex}
\input{a120-emRingerProtocol.tex}
\input{a130-molprobityProtocol.tex}
\input{a131-valCryoEMProtocol.tex}
\input{a140-realSpaceRefineProtocol.tex}
\input{a150-superposePdbsProtocol.tex}
\input{a160-dockInMapProtocol.tex}
\input{a170-exportToEMDB.tex}
\input{a180-asignOrigAndSampling.tex}








%\marginnote{swissmodel}[1cm]
%When predicting by homology a structure is constructed by aligning a target protein sequence with known template structures. The protein sequence can be obtained from many sources, for example NCBI or UniProt. The quality of a structure depends upon the similarity between the target sequence and the database sharing highest similarity is aligned. 

%Esto es otro parrafo


\end{appendices}

\end{document}

% NOTES
% to be implemented map2model (phenix)
%       chimera (see building parts of a protein without using a template)
